LibOS/shim/src/sys/shim_wait.c

```diff
@@ -117,15 +132,53 @@ pid_t shim_do_wait4(pid_t pid, int* status, int option, struct __kernel_rusage*
     unlock(&cur->lock);
 
 found:
-    if (status) {
+    if (infop) {
+        infop->si_pid = thread->tid;
+        infop->si_uid = thread->uid;
+        infop->si_signo = SIGCHLD;
+
         /* Bits 0--7 are for the signal, if any.
          * Bits 8--15 are for the exit code */
-        *status = thread->term_signal;
-        *status |= ((thread->exit_code & 0xff) << 8);
+        infop->si_code = thread->term_signal;
+        infop->si_code |= ((thread->exit_code & 0xff) << 8);
```

Comment:
This is not correct. Unlike the other `wait` functions, `si_code` contains the actual exit code unshifted if `si_status` is `CLD_EXITED`

Reply:
Looks like I also got code and status inverted in the bug report.

`si_code` should be one of the `CLD_` constants and `si_status` should be the signal number or exit code (unshifted).